pgmpy/inference/ApproxInference.py

```diff
@@ -196,3 +202,81 @@ def query(
         return self.get_distribution(
             samples, variables=variables, state_names=state_names, joint=joint
         )
+
+    def map_query(
+        self,
+        variables,
+        n_samples=int(1e4),
+        samples=None,
+        evidence=None,
+        virtual_evidence=None,
+        state_names=None,
+        show_progress=True,
+        seed=None,
+    ):
+        """
+        Computes the MAP Query using approx inference. Returns the
+        highest probable state in the joint distribution of `variables`.
+
+        Parameters
+        ----------
+        variables: list
+            List of variables for which the probability distribution needs to be calculated.
+
+        n_samples: int
+            The number of samples to generate for computing the distributions. Higher `n_samples`
+            results in more accurate results at the cost of more computation time.
+
+        samples: pd.DataFrame (default: None)
+            If provided, uses these samples to compute the distribution instead
+            of generating samples. `samples` **must** conform with the provided
+            `evidence` and `virtual_evidence`.
+
+        evidence: dict (default: None)
+            The observed values. A dict key, value pair of the form {var: state_name}.
+
+        virtual_evidence: list (default: None)
+            A list of pgmpy.factors.discrete.TabularCPD representing the virtual/soft
+            evidence.
+
+        state_names: dict (default: None)
+            A dict of state names for each variable in `variables` in the form {variable_name: list of states}.
+            If None, inferred from the data but is possible that the final distribution misses some states.
+
+        show_progress: boolean (default: True)
+            If True, shows a progress bar when generating samples.
+
+        seed: int (default: None)
+            Sets the seed for the random generators.
+
+        Examples
+        --------
+        >>> from pgmpy.utils import get_example_model
+        >>> from pgmpy.inference import ApproxInference
+        >>> model = get_example_model("alarm")
+        >>> infer = ApproxInference(model)
+        >>> print(infer.map_query(variables=["HISTORY", "CVP"]))
+        {'HISTORY': 'FALSE', 'CVP': 'NORMAL'}
+        """
+
```

Comment:
Please try to avoid unnecessary blank lines. Usually, throughout the codebase, we have tried to use blank lines to denote different stages (or blocks) or code.